MemoryMgr.GTA.h

```diff
@@ -20,7 +20,7 @@ namespace Memory
 		ptrdiff_t offset;
 	};
 
-	using AddrVariant = std::variant<uintptr_t, PatternAndOffset>;
+	using AddrVariant = std::variant<intptr_t, uintptr_t, PatternAndOffset>;
```

Comment:
I wonder if `uintptr_t` is needed here to begin with. If it's not, it'd simplify the `get_if` accesses quite a lot.

Reply:
I get errors like this without allowing `intptr_t` in AddrVariant:
```
/mnt/storagebeast/Development/General/SilentPatch/SilentPatchSA/AudioHardwareSA.cpp: At global scope:
/mnt/storagebeast/Development/General/SilentPatch/SilentPatchSA/AudioHardwareSA.cpp:6:65: error: no matching function for call to ‘AddressByVersion<void*>(int, int, int)’
    6 | static void* CAEDataStream__Initialise = AddressByVersion<void*>(0x4DC2B0, 0x4DC7A0, 0x4E7550);
      |                                          ~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/mnt/storagebeast/Development/General/SilentPatch/SilentPatch/Utils/MemoryMgr.GTA.h:385:10: note: candidate: ‘T AddressByVersion(Memory::AddrVariant, Memory::AddrVariant, Memory::AddrVariant) [with T = void*; Memory::AddrVariant = std::variant<unsigned int, Memory::PatternAndOffset>]’
  385 | inline T AddressByVersion(Memory::AddrVariant address10, Memory::AddrVariant address11, Memory::AddrVariant addressSteam)
      |          ^~~~~~~~~~~~~~~~
/mnt/storagebeast/Development/General/SilentPatch/SilentPatch/Utils/MemoryMgr.GTA.h:385:47: note:   no known conversion for argument 1 from ‘int’ to ‘Memory::AddrVariant’ {aka ‘std::variant<unsigned int, Memory::PatternAndOffset>’}
  385 | inline T AddressByVersion(Memory::AddrVariant address10, Memory::AddrVariant address11, Memory::AddrVariant addressSteam)
```

(I guess I could cast all of those constants but that would add extra complexity)

Reply:
That's the signed one, and I'm wondering if there's any point in keeping the unsigned variant.

Reply:
Removing the unsigned variant does work with MinGW GCC (but MSVC might be different)

Reply:
I will verify that locally on msvc later. This functionality only exists for GTA so there is no risk of breaking other games.